src/Lykke.Service.PayInvoice/Controllers/EmployeesController.cs

```diff
@@ -0,0 +1,208 @@
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Lykke.Service.PayInvoice.Controllers
+{
+    [Route("api")]
+    public class EmployeesController : Controller
+    {
+        private readonly IEmployeeService _employeeService;
+        private readonly ILog _log;
+
+        public EmployeesController(
+            IEmployeeService employeeService,
+            ILog log)
+        {
+            _employeeService = employeeService;
+            _log = log;
+        }
+        
+        /// <summary>
+        /// Returns merchant employees.
+        /// </summary>
+        /// <param name="merchantId">The merchant id.</param>
+        /// <returns>The collection of employees.</returns>
+        /// <response code="200">The collection of employees.</response>
+        [HttpGet]
+        [Route("merchants/{merchantId}/employees")]
+        [SwaggerOperation("EmployeesGetByMerchantId")]
+        [ProducesResponseType(typeof(List<EmployeeModel>), (int)HttpStatusCode.OK)]
+        public async Task<IActionResult> GetAsync(string merchantId)
+        {
+            IEnumerable<IEmployee> employees = await _employeeService.GetAsync(merchantId);
+
+            var model = Mapper.Map<List<EmployeeModel>>(employees);
+
+            return Ok(model);
+        }
+        
+        /// <summary>
+        /// Returns an employee.
+        /// </summary>
+        /// <returns>The employee.</returns>
+        /// <response code="200">The employee.</response>
+        /// <response code="404">Employee not found.</response>
+        [HttpGet]
+        [Route("merchants/{merchantId}/employees/{employeeId}")]
+        [SwaggerOperation("EmployeesGetByMerchantIdById")]
+        [ProducesResponseType(typeof(EmployeeModel), (int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetAsync(string merchantId, string employeeId)
+        {
+            try
+            {
+                IEmployee employee = await _employeeService.GetAsync(merchantId, employeeId);
+
+                var model = Mapper.Map<EmployeeModel>(employee);
+                
+                return Ok(model);
+            }
+            catch (EmployeeNotFoundException exception)
+            {
+                await _log.WriteWarningAsync(nameof(EmployeesController), nameof(GetAsync),
+                    merchantId.ToContext(nameof(merchantId))
+                        .ToContext(nameof(employeeId), employeeId)
+                        .ToJson(), exception.Message);
+                
+                return NotFound(ErrorResponse.Create(exception.Message));
+            }
+            catch (Exception exception)
+            {
+                await _log.WriteErrorAsync(nameof(EmployeesController), nameof(GetAsync),
+                    merchantId.ToContext(nameof(merchantId))
+                        .ToContext(nameof(employeeId), employeeId)
+                        .ToJson(), exception);
+
+                throw;
```

Comment:
Wouldn't it be better to return 500 explicitly?

Reply:
Middleware will do it and log request details.